sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/Common/SecurityHotspotTest.cs

```diff
@@ -0,0 +1,117 @@
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using FluentAssertions.Common;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using SonarAnalyzer.Common;
+using SonarAnalyzer.Helpers;
+using SonarAnalyzer.UnitTest.MetadataReferences;
+using SonarAnalyzer.UnitTest.PackagingTests;
+using SonarAnalyzer.UnitTest.Rules;
+using SonarAnalyzer.UnitTest.TestFramework;
+using SonarAnalyzer.Utilities;
+
+namespace SonarAnalyzer.UnitTest.Common
+{
+    [TestClass]
+    public class SecurityHotspotTest
+    {
+        [TestMethod]
+        public void SecurityHotspotRules_AreDisabledByDefault()
+        {
+            VerifyNoIssueReported(AnalyzerLanguage.CSharp);
+            VerifyNoIssueReported(AnalyzerLanguage.VisualBasic);
+        }
+
+        private static void VerifyNoIssueReported(AnalyzerLanguage language)
+        {
+            foreach (var analyzer in GetHotspotAnalyzers(language).Where(IsValid))
+            {
+                Verifier.VerifyNoIssueReported($"TestCases\\{GetTestCaseFileName(analyzer)}.{language.FileExtension}",
+                                               analyzer,
+                                               additionalReferences: GetAdditionalReferences());
+            }
+        }
+
+        private static string GetTestCaseFileName(DiagnosticAnalyzer analyzer)
+        {
+            var typeName = analyzer.GetType().Name;
+
+            return typeName switch
+            {
+                "ConfiguringLoggers" => "ConfiguringLoggers_AspNetCore",
+                "CookieShouldBeHttpOnly" => "CookieShouldBeHttpOnly_Nancy",
+                "CookieShouldBeSecure" => "CookiesShouldBeSecure_Nancy",
+                "DoNotHardcodeCredentials" => "DoNotHardcodeCredentials_DefaultValues",
+#if NETFRAMEWORK
+                "ExecutingSqlQueries" => "ExecutingSqlQueries_Net46",
+                "UsingCookies" => "UsingCookies_Net46",
+#else
+                "ExecutingSqlQueries" => "ExecutingSqlQueries_NetCore",
+                "UsingCookies" => "UsingCookies_NetCore",
+#endif
+                _ => typeName
+            };
+        }
+
+        private static bool IsValid(DiagnosticAnalyzer analyzer)
+        {
+#if NETFRAMEWORK
+            return true;
+#else
+            // IdentityModel is not available on .Net Core
+            return analyzer.GetType().Name != "ControllingPermissions";
+#endif
+        }
+
+        private static IEnumerable<SonarDiagnosticAnalyzer> GetHotspotAnalyzers(AnalyzerLanguage language) =>
```

Comment:
nitpick: in the logical order of reading, this and `IsSecurityHotspot` should be immediately after `VerifyNoIssueReported`

Reply:
I've reordered the methods a bit.